Example/CollectionViewToolsExample/FactoryExampleViewController.swift

```diff
@@ -0,0 +1,185 @@
+                       ImageData(image: #imageLiteral(resourceName: "nightlife-4"), description: "Foursh image description"),
+                       TextData(text: "Second topic", description: "Second topic description"),
+                       TextData(text: "Third topic", description: "Third topic description"),
+                       TextData(text: "Foursh topic", description: "Foursh topic description"),
+                       ImageData(image: #imageLiteral(resourceName: "nightlife-5"), description: "Fifth image description"),
+                       TextData(text: "Fifth topic", description: "Fifth topic description")]
+    
+    private lazy var unfoldedItemsFactory: CellItemFactory = {
+        let factory = AssociatedCellItemFactory<Any, TextCollectionViewCell>()
+        let sizeCell = TextCollectionViewCell()
+        factory.cellConfigurationHandler = { data, cell, cellItem in
+            let text: String
+            if let data = data as? ImageData {
+                text = data.description
+            }
+            else if let data = data as? TextData {
+                text = data.description
+            }
+            else {
+                text = ""
+            }
+            cell.titleLabel.text = text
+        }
+        factory.sizeConfigurationHandler = { data, collectionView, sectionItem in
+            let width = collectionView.bounds.width
+            let height = sizeCell.sizeThatFits(CGSize(width: width, height: .greatestFiniteMagnitude)).height
+            return CGSize(width: width, height: height + 48)
+        }
+        return factory
+    }()
+    
+    private lazy var cellItemFactory: CellItemFactory = {
+        let textCellItemFactory: CellItemFactory = {
+            let factory: AssociatedCellItemFactory<TextData, TextCollectionViewCell> = makeFactory()
+            
+            factory.cellConfigurationHandler = { data, cell, cellItem in
+                cell.titleLabel.text = data.text
+            }
+            
+            factory.sizeConfigurationHandler = { data, collectionView, sectionItem in
+                return CGSize(width: collectionView.bounds.width, height: 60)
+            }
+            
+            return factory
+        }()
+        
+        let imageCellItemFactory: CellItemFactory = {
+            let factory: AssociatedCellItemFactory<ImageData, ImageCollectionViewCell> = makeFactory()
+            
+            factory.cellConfigurationHandler = { data, cell, cellItem in
+                cell.imageView.image = data.image
+            }
+            
+            factory.sizeConfigurationHandler = { data, collectionView, sectionItem in
+                let width = collectionView.bounds.width
+                let aspectRatio = data.image.size.width / data.image.size.height
+                return CGSize(width: width, height: width / aspectRatio)
+            }
+            
```

Comment:
> What's about another functions like highlighting or selecting?
`cellItem` is still available from `cellConfigurationHandler` and `cellItemConfigurationHandler` to provide specific selecting. Moreover it is possible to provide any custom cellItem in the `initializationHandler` instead of `UniversalCellItem`